core/src/main/java/org/elasticsearch/search/aggregations/pipeline/derivative/InternalDerivative.java

```diff
@@ -89,4 +90,17 @@ public XContentBuilder doXContentBody(XContentBuilder builder, Params params) th
         }
         return builder;
     }
+
+    @Override
+    protected int doHashCode() {
+        return Objects.hash(normalizationFactor, super.doHashCode());
+    }
+
+    @Override
+    protected boolean doEquals(Object obj) {
+        InternalDerivative other = (InternalDerivative) obj;
+        return Objects.equals(getValue(), other.getValue())
+                && Objects.equals(normalizationFactor, other.normalizationFactor)
+                && Objects.equals(format.getWriteableName(), other.format.getWriteableName());
```

Comment:
Yep you are right, I thought that `IntenalSimpleValue.doEquals(Object)` was being called here but it isn't